main.go

```diff
@@ -32,48 +96,317 @@ func main() {
+	// if err != nil {
+	// 	log.Fatal(err)
+	// }
+	// defer rows.Close()
 
-	for rows.Next() {
-		var id int
-		var email string 
-		var state string
+	// for rows.Next() {
+	// 	var id int
+	// 	var email string
+	// 	var state string
 
-		if err := rows.Scan(&id, &email, &state); err != nil {
-			log.Fatal(err)
-		}
+	// 	if err := rows.Scan(&id, &email, &state); err != nil {
+	// 		log.Fatal(err)
+	// 	}
 
-		fmt.Println(id, email, state)
-	}
+	// 	fmt.Println(id, email, state)
+	// }
 
-	if err := rows.Err(); err != nil {
-		log.Fatal(err)
-	}
+	// if err := rows.Err(); err != nil {
+	// 	log.Fatal(err)
+	// }
 
 	app := &cli.App{
 		Name: "store",
 		Commands: []*cli.Command{
 			{
-				Name:  "example",
-				Usage: "show a bunch of stuff",
+				Name:  "create-customer",
+				Usage: "Creates a new customer to go in the customers database, must specify email and state(2 letter code)",
+				Action: func(cCtx *cli.Context) error {
+					if cCtx.NArg() < 2 {
+						return errors.New("Must specify email and state")
+					}
+					email := cCtx.Args().Get(0)
+					state := cCtx.Args().Get(1)
+					if len(state) != 2 {
+						return errors.New("State length must be 2")
+					}
+					_, ok := states[strings.ToUpper(state)]
+					if !ok {
+						return errors.New("State must be a valid U.S. State or Territory")
+					}
+
+					insertStatement := "INSERT INTO Customers (email, state) VALUES (?, ?)"
+					res, err := db.Exec(insertStatement, email, state)
+
+					if err != nil {
+						return err
+					}
+					customerID, err := res.LastInsertId()
+					if err != nil {
+						return err
+					}
+					w := tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', tabwriter.AlignRight|tabwriter.Debug)
+					fmt.Fprintln(w, "Customer_id\tEmail\tState\t")
+					fmt.Fprintf(w, "%d\t%s\t%s\t\n", customerID, email, state)
+					w.Flush()
+
+					return nil
+				},
+			},
+			{
+				Name:  "create-product",
+				Usage: "Creates a new product to go in the products database, must specify name and price",
+				Flags: []cli.Flag{
+					&cli.StringFlag{
+						Name:  "sku",
+						Usage: "the sku of the product",
+					},
+				},
+				Action: func(cCtx *cli.Context) error {
+					if cCtx.NArg() < 2 {
+						return errors.New("Must specify name and price")
+					}
+
+					name := cCtx.Args().Get(0)
+					price, err := strconv.Atoi(cCtx.Args().Get(1))
+					if err != nil {
+						return err
+					}
+					sku := cCtx.String("sku")
+					if sku != "" {
+						insertStatement := "INSERT INTO Products (name, price, sku) VALUES (?, ?, ?)"
+						res, err := db.Exec(insertStatement, name, price, sku)
+						if err != nil {
+							return err
+						}
+
+						productId, err := res.LastInsertId()
+						if err != nil {
+							return err
+						}
+						w := tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', tabwriter.AlignRight|tabwriter.Debug)
+						fmt.Fprintln(w, "Product_id\tName\tPrice\tSku\t")
+						fmt.Fprintf(w, "%d\t%s\t%v\t%v\t\n", productId, name, price, sku)
+						w.Flush()
```

Comment:
Notice how lines 181-193 are almost exactly the same as lines 196-207. Can we condense this logic into a single function?

hint: look at the differences between 181-193 and 196-207: 
  1 .The `insertStatement` string
  2. The arguments going into `db.Exec`. 
  
 The differences should be the parameters to a helper function. 
 
 Also note that line 205 is incorrect: we should always print the `sku` column even if it isn't present:  https://github.com/vivek-shah-13/store/#we-should-have-show-commands-for-customers-products-and-orders
 ```
  > store show-products
  PRODUCT_ID        NAME       PRICE       SKU
  1                 Bananas     3.99       aa1
  2                 Milk        4.50       
  3                 Cookies     1.99       ab2

  > store show-products --name "M%"
  PRODUCT_ID        NAME       PRICE       SKU
  2                 Milk        4.50       
  5                 Markers     8.99       xi1    
  ```

Reply:
https://go.dev/play/p/D8E3YnQo8tq